src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int MAX_SIZE = 50;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_COUNT)) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE) + 1);
```

Comment:
i would recommend to move `random.nextInt(MAX_SIZE) + 1)` logic to a private method, if you decide to change it to `random.nextInt(MAX_SIZE) + 2` you will need to do it in many places, instead of one